src/server/scripts/EasternKingdoms/ScarletEnclave/chapter1.cpp

```diff
@@ -883,6 +884,60 @@ class npc_ros_dark_rider : public CreatureScript
 
 };
 
+enum GiftOfTheHarvester
+{
+    SPELL_GIFT_OF_THE_HARVESTER = 52479,
+    SPELL_MINER_GHOUL_TRANSFORM = 52490,
+    SPELL_MINER_GHOST_TRANSFORM = 52505
+};
+
+struct gift_of_the_harvesterAI : public GameObjectAI
+{
+    gift_of_the_harvesterAI(GameObject* go) : GameObjectAI(go) { }
+
+    void Reset() override
+    {
+        _scheduler.Schedule(2s, [this](TaskContext /*context*/)
+        {
+            me->SendCustomAnim(0);
+            me->CastSpell(nullptr, SPELL_GIFT_OF_THE_HARVESTER, true);
```

Comment:
This spell has TARGET_UNIT_DEST_AREA_ENTRY
Has any conditions for this spell in db?

Reply:
missing _scheduler.CancellAll on death/removefromworld maybe?

Reply:
You think this might be the reason of crashes?

Reply:
maybe it is needed? (before scheduler)

```cpp
_scheduler.SetValidator([this] { return me != nullptr; });
```

Looks like when schedule is running, object is despawned already.
So, looks like 2s time is wrong, or object need more time before despawn.

Reply:
It's not "me" being null

Reply:
me = caster
and caster is null at this point: https://gist.github.com/Sorikoff/3d1ed6ee2e5573198fb99c8c1ebaf940#file-gistfile1-txt-L17
So while this schedule is executing, object (me/caster) will despawn.

Or, idk if objects still use a little trigger creature to cast spells?!

Reply:
>#17 0x0000555557b81f42 in WorldObject::CastSpell (this=0x7fff82615c00, target=0x0, spellId=52479

"this" is "me" and it's not null

Reply:
ok... but why caster is null in #2?
if i remember right, all objects use a world trigger to cast spells, idk if is current behavior still.

Reply:
Maybe it's a bug when the caster is a gameobject